functions/arrow-functions.md

```diff
@@ -0,0 +1,239 @@
+| `'hey there.'`                                      | In one-line arrow functions, we can exclude the curly braces and the `return` keyword. However, this expression **_must_** be on the same line as the rest of the function. |
+| `;`                                                 | This statement now must conclude with a semicolon.                                                                                                                          |
+
+<!-- ### Exercise
+
+Convert this into an arrow function and minimize the code typed.
+
+```javascript
+const hello = function () {
+  return "hello world";
+};
+``` -->
+
+## Syntax with Parameters
+
+### With Multiple Parameters or Zero Parameters
+
+Like "normal" functions, arrow functions have their parameters listed in the parentheses.
+
+With zero parameters, we leave the parentheses empty:
+
+```javascript
+const sayHelloWorld = () => "Hello World!";
+```
+
+With multiple parameters, they are comma-separated.
+
+```javascript
+const add = (x, y) => x + y;
+```
+
+### With One Parameter
+
+This arrow function takes a single number and returns that number squared:
+
+```javascript
+const squareNum = (number) => {
+  return number * number;
+};
+console.log(squareNum(3)); // 9
+```
+
+When we have an arrow function that takes in _one and only one_ parameter, the parentheses around the single parameter are optional.
+
+Consider this refactoring:
+
+```javascript
+const squareNum = (number) => {
+  return number * number;
+};
+console.log(squareNum(3)); // 9
+```
+
+If we don't add parentheses when there are zero or more than one parameter, we'll get a syntax error!
+
+### !callout-info
+
+## Bonus Refactor: One Parameter and One Line
+
+We can refactor our `squareNum` arrow function, taking advantage of its one parameter and one-line return:
+
+<!-- prettier-ignore-start -->
+```js
+const squareNum = number => number * number;
+```
+<!-- prettier-ignore-end -->
+
+### !end-callout
+
+## One-Line Arrow Functions Returning Objects
+
+Consider this `makePerson` function, whose responsibility is to return an object.
+
+```javascript
+const makePerson = function (id, name) {
+  return {
+    id: id,
+    name: name,
+  };
+};
+```
+
+When we refactor it to a one-line arrow function, _because it specifically returns an object_, we need to wrap our object in parentheses.
+
+```javascript
+const makePerson = (id, name) => ({ id: id, name: name });
+```
+
+In general, wrapping a value in parentheses is a great way to ensure it's not interpreted as a function body.
+
+### !callout-info
+
+## JavaScript's Object Shorthand
+
+In JavaScript, we'll make many object literals over time. Imagine a situation when our object literal needs a key-value pair. When we have **exactly** a _variable_ that references the _value_ we want to include **_and_** that variable is named _exactly_ the name of the key we're inserting, we can use object literal shorthand.
```

Comment:
remove the portion that was moved up into the suggested paragraph above
more consistent formatting on the emphasized elements
```suggestion
When we have **exactly** a _variable_ that references the _value_ we want to include **_and_** that variable is named **exactly** the name of the key we're inserting, we can use object literal shorthand.
```